### PR TITLE
fix: move metrics setting to `_check_duty`

### PR DESCRIPTION
**Pull request by vgorkavenko**

### CODE REVIEW

File: src/modules/csm/checkpoint.py [hunk 1]
```diff
@@ -202,6 +204,9 @@ def _check_duty(
             self.state.add_processed_epoch(duty_epoch)
             self.state.commit()
             self.state.status()
+            unprocessed_epochs = self.state.unprocessed_epochs
+            CSM_UNPROCESSED_EPOCHS_COUNT.set(len(unprocessed_epochs))
+            CSM_MIN_UNPROCESSED_EPOCH.set(min(unprocessed_epochs or {EpochNumber(0)}))
```
Comment (madlabman) [inline src/modules/csm/checkpoint.py:209]: I'd wrap -1 instead.

File: src/modules/csm/checkpoint.py [hunk 2]
```diff
@@ -202,6 +204,9 @@ def _check_duty(
             self.state.add_processed_epoch(duty_epoch)
             self.state.commit()
             self.state.status()
+            unprocessed_epochs = self.state.unprocessed_epochs
```
Comment (madlabman) [inline src/modules/csm/checkpoint.py:207]: Why so?

Comment (vgorkavenko) [reply src/modules/csm/checkpoint.py:207]: to do calcs once and be sure that the value will be the same for both of metrics

File: src/modules/csm/checkpoint.py [hunk 3]
```diff
@@ -202,6 +204,9 @@ def _check_duty(
...
+            CSM_UNPROCESSED_EPOCHS_COUNT.set(len(unprocessed_epochs))
+            CSM_MIN_UNPROCESSED_EPOCH.set(min(unprocessed_epochs or {EpochNumber(-1)}))
```
Comment (F4ever) [inline src/modules/csm/checkpoint.py:209]: Maybe better
```suggestion
            if unprocessed_epochs:
                CSM_MIN_UNPROCESSED_EPOCH.set(min(unprocessed_epochs))
            else:
                CSM_MIN_UNPROCESSED_EPOCH.set(max(processed_epochs) + 1))
```

Comment (vgorkavenko) [reply src/modules/csm/checkpoint.py:209]: it will mislead

**Review by F4ever — APPROVED** [review]

File: src/modules/csm/checkpoint.py (hunk 2 above)
Comment (F4ever) [reply src/modules/csm/checkpoint.py:207]: Let's do it outside lock?

Comment (F4ever) [reply src/modules/csm/checkpoint.py:207]: self.state.status() seems can be done outside

**Review by F4ever — APPROVED** [review]

**Pull request merged**